### ISSUE TITLE
web3j.replayPastAndFutureBlocksFlowable issue

### ISSUE THREAD

**Issue by abhaygupta1999** [web3j/web3j#1877]
# _Issue_title_
web3j.replayPastAndFutureBlocksFlowable keeps on running on blockingSubscribe but there was no active web3j filter running at that time. So, basically, it means an empty flowable is running.

## _Issue_description_
So, I am using this code in my service:

while (true) {
      try {
        web3j.replayPastAndFutureBlocksFlowable(blockNumber, true)
        .blockingSubscribe(ethBlock ->
                        processBlock(ethBlock.getBlock()),
                        log::error
                    );
          blockNumber = getLastBlockProcessed();
          log.info("Resuming transactions processing from block number: {}", blockNumber);
      } catch (Exception e) {
        log.error("Some error occurred while getting the block due to: ", e);
        blockNumber = getLastBlockProcessed();
        log.info("Resuming transactions processing from block number: {}", blockNumber);
      }

And I am using Alchemy HTTP, and I got one invalid request in alchemy as {"jsonrpc":"2.0","id":265486,"error":{"code":-32000,"message":"filter not found"}} for **eth_getFilterChanges**.

Can anyone please help me in this?

**Comment by gtebrean** [comment]
was this happening before version 4.9.7?

**Comment by abhaygupta1999** [comment]
Yes, I was using 4.8.7 version. Any changes regarding this in 4.9.7? @gtebrean

**Comment by gtebrean** [comment]
Yes there are 10 versions between them and in one of them filter type has been updated I will check and come back with an example.
